builder/oracle/oci/config.go

```diff
@@ -77,7 +77,8 @@ type Config struct {
 	UserDataFile string `mapstructure:"user_data_file"`
 
 	// Networking
-	SubnetID string `mapstructure:"subnet_ocid"`
+	SubnetID          string                 `mapstructure:"subnet_ocid"`
+	CreateVnicDetails map[string]interface{} `mapstructure:"create_vnic_details"`
```

Comment:
I think we can get all of the mapping for "free" via our normal decode operations if instead of defining a map[string]interface{} we define a separate struct with all the valid fields for create_vnic_details. 

example: 

```
type CreateVNICDetails struct {
    AssignPublicIP bool `mapstructure:"assign_public_ip" required:"false"
    DisplayName string `mapstructure:"display_name" required:"false"
    HostnameLabel string `mapstructure:"hostname_label" required:"false"
    NSGIDs []string `mapstructure:"nsg_ids" required:"false"
    ... etc etc
}
```

Then you add this into the config as: 

```
type Config struct {
       // blah blah blah all the other settings
 	CreateVnicDetails CreateVNICDetails `mapstructure:"squash"`
      // blah blah
}
```

Reply:
Thanks. I had considered this approach, but thought I would still have to do a bunch of type assertions/conversions. Happy to see there's some magic in the background making this pretty straight forward.